### ISSUE TITLE
Consider dropping `nnunet-neuropoly` fork

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#5274]
Follow up to:

- https://github.com/MIC-DKFZ/nnUNet/issues/3025

The fork was created under the impression that our needs could only be met by maintaining the fixes ourselves. But, in the past year, 2/3 of our fixes have been merged upstream:

- [x] torch<2.3 support
- [x] custom trainer support
- [ ] return logits per fold for more control over averaging process

If we can get the last fix merged upstream, we'll be able to drop the fork and continue working with the nnunet devs to make sure our needs are met.
